src/main/java/br/com/matheus/agendamentoconsultas/service/MedicoService.java

```diff
@@ -79,12 +80,14 @@ public Medico cadastrar(final RequestCadastroMedicoDTO medicoDTO) {
                         enderecoDTO.cidade(), enderecoDTO.uf(), enderecoDTO.cep()))
                 .especializacao(Especializacao.valueOf(medicoDTO.especializacao()))
                 .build();
-        Set<HorarioAtendimento> horariosAtendimento = medicoDTO.horariosAtendimento().stream()
-                .map(hrAt -> new HorarioAtendimento(new HorarioAtendimentoPK(medico.getId(),
-                        DiaDaSemana.valueOf(hrAt.diaDaSemana())), medico,
-                        LocalTime.parse(hrAt.horaInicial()), LocalTime.parse(hrAt.horaFinal())))
-                .collect(Collectors.toSet());
-        medico.setHorariosAtendimento(horariosAtendimento);
+
+        medicoDTO.horariosAtendimento().forEach(hrAt -> {
+            DiaDaSemana diaDaSemana = DiaDaSemana.valueOf(hrAt.diaDaSemana());
+            LocalTime horaInicial = LocalTime.parse(hrAt.horaInicial());
+            LocalTime horaFinal = LocalTime.parse(hrAt.horaFinal());
+            medico.adicionaHorarioAtendimento(diaDaSemana, horaInicial, horaFinal);
```

Comment:
Sugestao de encapsulamento, para não precisar converter em todo caso de uso

```suggestion
            DiaDaSemana diaDaSemana = hrAt.diaDaSemanaAsEnum();
            LocalTime horaInicial = hrAt.horaInicialAsLocalTime();
            LocalTime horaFinal = hrAt.horaFinalAsLocalTime();
            medico.adicionaHorarioAtendimento(diaDaSemana, horaInicial, horaFinal);
```

Reply:
@bugorin Sugestão aceita. Dessa forma foi possivel adicionar métodos nos records de DTO que facilitaram bastante a utilização dos dados.